src/repo.rs

```diff
@@ -9,15 +9,24 @@ const ENV_SSH_PASSPHRASE: &str = "SSH_PASSPHRASE";
 
 pub(crate) struct RepoManager {
     pub(crate) temp_dir: TempDir,
-    pub(crate) ssh_key_path: PathBuf,
+    pub(crate) ssh_key_path: Option<PathBuf>,
 }
 
 impl RepoManager {
+    pub(crate) fn new() -> Result<Self, io::Error> {
```

Comment:
Hi octoaide
It doesn't seem necessary to have two separate constructor methods (new and new_with_key) for RepoManager. Could you revise the code to use a single constructor method in the form of pub(crate) fn new(ssh_key_path: Option<PathBuf>) -> Result<Self, io::Error>?